crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -6,35 +6,46 @@ use mqtt_channel::StreamExt;
 use mqtt_channel::Topic;
 use mqtt_channel::TopicFilter;
 use reqwest::Url;
+use std::collections::HashMap;
 use std::time::Duration;
 use tedge_config::mqtt_config::MqttConfigBuildError;
 use tedge_config::TEdgeConfig;
 use tracing::error;
 use tracing::info;
 
+#[derive(thiserror::Error, Debug)]
+pub enum C8yEndPointError {
+    #[error("Internal id is empty")]
+    EmptyInternalID,
```

Comment:
Such an error message makes no sense for the end-user.

I would be better to get  rid of this error case and even error type. These are more a sign that the struct design is wrong (No internal should be encoded by `None`) and in any case the true error is more about "failing to get the c8y identifier for the device".

Reply:
Refactored the error message and type. Renamed the type to `InternalIdNotFound`